tests/oval_tester.py

```diff
@@ -0,0 +1,153 @@
+        skeleton = header + ssg.constants.oval_footer
+        root = ssg.xml.ElementTree.fromstring(skeleton.encode("utf-8"))
+        root.append(definitions)
+        root.append(tests)
+        root.append(objects)
+        root.append(states)
+        if list(variables):
+            root.append(variables)
+        id_translator = ssg.id_translate.IDTranslator("test")
+        root_translated = id_translator.translate(root)
+
+        ssg.xml.ElementTree.ElementTree(root_translated).write(oval_path)
+
+    def _get_result(self, oscap_output):
+        pattern = re.compile(
+            r"^Definition oval:[A-Za-z0-9_\-\.]+:def:[1-9][0-9]*: (\w+)$")
+        for line in oscap_output.splitlines():
+            matched = pattern.match(line)
+            if matched:
+                return matched.group(1)
+        return None
+
+    def _create_config_file(self, config_file_content, tmp_dir):
+        config_file_path = os.path.join(tmp_dir, "config")
+        if config_file_content:
+            with open(config_file_path, "w") as f:
+                f.write(config_file_content)
+        return config_file_path
+
+    def _create_oval(self, oval_content, config_file_path, tmp_dir):
+        oval_content = oval_content.replace("CONFIG_FILE", config_file_path)
+        shorthand_path = os.path.join(tmp_dir, "shorthand")
+        with open(shorthand_path, "w") as f:
+            f.write(oval_content)
+        oval_path = os.path.join(tmp_dir, "oval.xml")
+        self._expand_shorthand(shorthand_path, oval_path, self.mock_env_yaml)
+        return oval_path
+
+    def _evaluate_oval(self, oval_path):
+        results_path = oval_path + ".results.xml"
+        oscap_command = [
+            "oscap", "oval", "eval", "--results", results_path, oval_path]
+        oscap_process = subprocess.Popen(
+            oscap_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        oscap_stdout, oscap_stderr = oscap_process.communicate()
+        def_result = self._get_result(oscap_stdout.decode("utf-8"))
+        return def_result, results_path
+
+    def test(self, description, oval_content, config_file_content,
+             expected_result):
+        """
+        Execute a test.
+        description: a very short description to be displayed in test output
+        oval_content: content of the OVAL shorthand file written in a way you
+                      write OVALs in SSG rules (not a valid OVAL)
+        config_file_content: content of the text configuration file that the
+                             OVAL will check
+        expected_result: expected result of evaluation of the OVAL definition
+        """
+        try:
+            tmp_dir = tempfile.mkdtemp()
+            config_file_path = self._create_config_file(
+                config_file_content, tmp_dir)
+            oval_path = self._create_oval(
+                textwrap.dedent(oval_content), config_file_path, tmp_dir)
```

Comment:
I have just this dedent-related remark - please dedent test cases only where they are defined. That way, one can get rid of those `\n` in strings.
The problem with dedenting inside the framework is that one may forget about it, and then the test case definition will look differently from what is actually tested.